Samples/Client/SwiftClient/SwiftClient/VideoStreamViewController.swift

```diff
@@ -0,0 +1,640 @@
+        return peerListDisplay[row]
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        if peerListDisplay.count > 1 && row != 0 {
+            print(peerListDisplay[row])
+            let peerListDisplayString = peerListDisplay[row]
+            let endPeerIdIndex = peerListDisplayString.range(of: ",", options: .literal)!.lowerBound
+            let startPeerIdIndex = peerListDisplayString.startIndex
+            
+            let peerId = Int(peerListDisplayString[startPeerIdIndex..<endPeerIdIndex])!
+            joinPeer(peerId: peerId)
+        }
+    }
+}
+
+extension VideoStreamViewController : RTCPeerConnectionDelegate {
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didChange stateChanged: RTCSignalingState) {}
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didChange newState: RTCIceGatheringState) {}
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didRemove stream: RTCMediaStream) {}
+    public func peerConnectionShouldNegotiate(_ peerConnection: RTCPeerConnection) {}
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didChange newState: RTCIceConnectionState) {}
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didRemove candidates: [RTCIceCandidate]) {}
+    
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didAdd stream: RTCMediaStream) {
+        print("new stream for a remote peer")
+        videoStream = stream
+        if let videoTrack = stream.videoTracks.last {
+            videoTrack.add(renderView)
+            
+            DispatchQueue.main.async {
+                self.view.addSubview(self.renderView)
+            }
+        }
+    }
+    
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didGenerate candidate: RTCIceCandidate) {
+        print("new ice candidate found")
+        let jsonData: [String: String] = [
+            "sdpMLineIndex" : String(candidate.sdpMLineIndex),
+            "sdpMid": candidate.sdpMid!,
+            "candidate": candidate.sdp
+        ]
+        self.sendToPeer(peerId: newIceCandidatePeerId, data: self.convertDictionaryToData(dict: jsonData)!)
+    }
+    
+    public func peerConnection(_ peerConnection: RTCPeerConnection, didOpen dataChannel: RTCDataChannel) {
+        print("didOpenChannel has been called, (js equiv = ondatachannel")
+        inputChannel = dataChannel
+        inputChannel.delegate = self
+    }
+}
+
+extension VideoStreamViewController {
+    func matCreate() -> [CGFloat] {
+        let out: [CGFloat] = [ 1.0, 0.0, 0.0, 0.0,   0.0, 1.0, 0.0, 0.0,  0.0, 0.0, 1.0, 0.0,  0.0, 0.0, 0.0, 1.0]
+        return out
+    }
+    
+    func matClone(a: [CGFloat]) -> [CGFloat] {
+        var out: [CGFloat] = [ 1.0, 0.0, 0.0, 0.0,   0.0, 1.0, 0.0, 0.0,  0.0, 0.0, 1.0, 0.0,  0.0, 0.0, 0.0, 1.0]
```

Comment:
any reason we need to initialize this manually? and if so, can we use `matCreate` to simplify this work?

Reply:
All of the math code is directly analogous to mathmatrix.js. There are many places that it could be simplified (for instance using for loops), but I left it looking nearly identical to the web client for code clarity purposes when people compare the implementations. I can make this code more "swifty" (and modular) of you'd like.

Reply:
i wobble on this - what do you think? being my swift experience is limited, if this feels uniquely not swifty (to the point that a swift dev would be sad) perhaps the updates would be good (and we'd comment them) - i'd like to see these clients feel more like experiences built for the language and the platform they're in rather than ports, but i know we aren't there across the board today.